rackcity/utils/change_planner_utils.py

```diff
@@ -289,13 +298,69 @@ def get_page_count_response_for_cp(request, change_plan):
     return JsonResponse({"page_count": page_count})
 
 
-def get_changes_on_asset(asset, assetCP):
+def network_connections_have_changed(asset, asset_cp):
+    network_ports_cp = NetworkPortCP.objects.filter(asset=asset_cp)
+    network_ports = NetworkPort.objects.filter(asset=asset)
+    for network_port_cp in network_ports_cp:
+        # Get NetworkPort connected to this port live
+        network_port = network_ports.get(
+            port_name=network_port_cp.port_name,
+        )
+        connected_port_live = network_port.connected_port
+        # Get NetworkPort associated with the NetworkPortCP connected to this port live
+        cp_connected_port_in_cp = network_port_cp.connected_port
+        real_connected_port_in_cp = None
+        if cp_connected_port_in_cp:
+            related_asset = cp_connected_port_in_cp.asset.related_asset
+            if related_asset:
+                try:
+                    real_connected_port_in_cp = NetworkPort.get(
+                        asset=related_asset,
+                        port_name=cp_connected_port_in_cp.port_name,
+                    )
+                except ObjectDoesNotExist:
+                    real_connected_port_in_cp = None
+        # Check for match
+        if connected_port_live != real_connected_port_in_cp:
+            return True
+    return False
```

Comment:
this took some crazy brain power and drawings to get right I think it's right now it looks like it's working thank you and goodbye @clairecfu

Reply:
I think you could have just checked the port name and hostname but go off queen

Reply:
oh


my






god

Reply:
except hostnames & asset numbers could have changed in change plan so i dont love that for identifying an asset actually. I stand by my long and winding path

Reply:
One could make the argument that if one disconnected host1 from host2 and connected it to host3 but then renamed host2 to host3 and renamed host3 to host2 that they actually did nothing at all

Reply:
But I stand with you.